consensus/src/consensus/remote_event_dispatcher.rs

```diff
@@ -0,0 +1,312 @@
+
+use nimiq_blockchain::Blockchain;
+use nimiq_blockchain_interface::AbstractBlockchain;
+use nimiq_blockchain_interface::BlockchainEvent;
+use nimiq_keys::Address;
+use nimiq_network_interface::request::Handle;
+use nimiq_network_interface::{network::Network, request::request_handler};
+
+use crate::messages::{PushAdressNotification, RequestSubscribeToAddress};
+use crate::messages::{ResponseRequestTransactionsByAddress, TxAddressSubscriptionOperation};
+use crate::SubscribebyAdressErrors::*;
+
+//The max number of peers that can be subscribed.
+pub const MAX_SUBSCRIBED_PEERS: usize = 5;
+//The max number of addresses, per peer, that can be subscribed
+pub const MAX_SUBSCRIBED_PEERS_ADDRESSES: usize = 10;
+
+impl<N: Network>
+    Handle<N, ResponseRequestTransactionsByAddress, Arc<RwLock<RemoteEventDispatcherState<N>>>>
+    for RequestSubscribeToAddress
+{
+    fn handle(
+        &self,
+        peer_id: N::PeerId,
+        state: &Arc<RwLock<RemoteEventDispatcherState<N>>>,
+    ) -> ResponseRequestTransactionsByAddress {
+        match self.operation {
+            TxAddressSubscriptionOperation::Initiate => {
+                if state.read().contains_peer(&peer_id) {
+                    return ResponseRequestTransactionsByAddress {
+                        result: Err(InvalidOperation),
+                    };
+                }
+                if state.read().number_of_peers() > MAX_SUBSCRIBED_PEERS {
+                    return ResponseRequestTransactionsByAddress {
+                        result: Err(TooManyPeers),
+                    };
+                }
+
+                if self.addresses.len() > MAX_SUBSCRIBED_PEERS_ADDRESSES {
+                    return ResponseRequestTransactionsByAddress {
+                        result: Err(TooManyAddresses),
+                    };
+                }
+
+                state.write().add_peer(&peer_id, self.addresses.clone());
+            }
+            TxAddressSubscriptionOperation::Update => {
+                if !state.read().contains_peer(&peer_id) {
+                    return ResponseRequestTransactionsByAddress {
+                        result: Err(InvalidOperation),
+                    };
+                }
+                if self.addresses.len() > MAX_SUBSCRIBED_PEERS_ADDRESSES {
+                    return ResponseRequestTransactionsByAddress {
+                        result: Err(TooManyAddresses),
+                    };
+                }
+                state.write().update_peer(&peer_id, self.addresses.clone());
+            }
+            TxAddressSubscriptionOperation::Terminate => {
+                if !state.read().contains_peer(&peer_id) {
+                    return ResponseRequestTransactionsByAddress {
+                        result: Err(InvalidOperation),
+                    };
+                }
+
+                state.write().remove_peer(&peer_id);
+            }
+        }
+        ResponseRequestTransactionsByAddress { result: Ok(()) }
```

Comment:
Usually subscriptions return a subscription ID that can then be used to manage that subscription. Did you decide against that pattern on purpose?